src/lib.rs

```diff
@@ -3133,6 +3135,64 @@ pub trait Itertools: Iterator {
         self.k_largest_by(k, k_smallest::key_to_cmp(key))
     }
 
+    /// Consumes the iterator and return an iterator of the last `n` elements.
+    ///
+    /// The iterator, if directly collected to a `VecDeque`, is converted
+    /// without any extra copying or allocation cost.
+    /// If directly collected to a `Vec`, it may need some data movement
+    /// but no re-allocation.
+    ///
+    /// ```
+    /// use itertools::{assert_equal, Itertools};
+    ///
+    /// let v = vec![5, 9, 8, 4, 2, 12, 0];
+    /// assert_equal(v.iter().tail(3), &[2, 12, 0]);
+    /// assert_equal(v.iter().tail(10), &v);
+    ///
+    /// assert_equal((0..100).tail(10), 90..100);
+    ///
+    /// assert_equal((0..100).filter(|x| x % 3 == 0).tail(10), (72..100).step_by(3));
+    /// ```
+    ///
+    /// For double ended iterators without side-effects, you might prefer
+    /// `.rev().take(n).rev()` to have a similar result (lazy and non-allocating)
+    /// without consuming the entire iterator.
+    #[cfg(feature = "use_alloc")]
+    fn tail(self, n: usize) -> VecDequeIntoIter<Self::Item>
+    where
+        Self: Sized,
+    {
+        match n {
+            0 => {
+                self.last();
+                VecDeque::new()
+            }
+            1 => self.last().into_iter().collect(),
+            _ => {
+                // Skip the starting part of the iterator if possible.
+                let (low, _) = self.size_hint();
+                let mut iter = self.fuse().skip(low.saturating_sub(n));
```

Comment:
I hope I did not miss relevant discussion, but is it ok here if `size_hint` is incorrect? E.g. an iterator with 3 elements, but `size_hint` returns "at least 4": `tail(2)` would then return 1 element, right?

Reply:
I had similar surprise myself (I asked for similar usage [here](https://github.com/rust-itertools/itertools/pull/899#discussion_r1526853345)) but the principal usage of `size_hint` outside memory allocation is `ExactSizeIterator::len` which we consider to be correct, right?

The [docs](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint) mentions that the size hint should not be trusted if an incorrect implementation might lead to memory safety violation. Other than that, an incorrect implementation is "a violation of the trait’s protocol" just as any other incorrect specialization I would say.